### PR TITLE
feat(utils/function) :pick function

### PR DESCRIPTION
**Pull request by vidyabhandari**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `pick` utility function for selecting specific properties from objects.
	- Added a `Pick` component that utilizes the `pick` function to extract user information from a nested object.
- **Bug Fixes**
	- Comprehensive unit tests implemented for the `pick` function to ensure correct behavior across various scenarios, including edge cases.

### CODE REVIEW

File: package/src/utils/function/pick/pick.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/** 
+* Creates an object composed of the picked object properties.
+* @param {object} obj - Source object to pick properties from
+* @param {string[]} keys - The key to pick from the string
+* @returns {object} - Return the new object
+*/
+
+export const pick = (obj: { [key: string]: any }, keys: string[]): { [key: string]: any } => {
+    const result: { [key: string]: any } = {};
+
```
Comment (bharat-dussa) [inline package/src/utils/function/pick/pick.ts:9]: handle for params validations. 
User can pass any data type they want, we have restrict that

File: package/src/utils/function/pick/pick.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    keys.forEach(key => {
```
Comment (bharat-dussa) [inline package/src/utils/function/pick/pick.ts]: use for of

File: package/src/utils/function/pick/pick.ts (hunk 1 above)
Comment (bharat-dussa) [reply package/src/utils/function/pick/pick.ts:9]:
```suggestion
if (!obj || typeof obj !== "object") {
    throw new Error("The first argument must be a valid object");
  }

  if (!Array.isArray(keys)) {
    throw new Error("The second argument must be an array of strings");
  }
```

**Review by bharat-dussa — CHANGES_REQUESTED** [review]

File: package/src/utils/function/pick/pick.ts (hunk 2 above)
Comment (bharat-dussa) [reply package/src/utils/function/pick/pick.ts]:
```suggestion
      for (const key of keys)
```

File: smart-app/src/components/function/pick.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        user:{
+            name:'Ram',
+            age: 20,
+            address:{
+                state:'Maharastra',
+                city:'Mumbai',
+                pincode:'421302'
+            }
+        }
+    };
```
[the remark this answers is not in the document]
Comment (bharat-dussa) [reply smart-app/src/components/function/pick.tsx:15]: @coderabbitai smart-app repo is for testing, this is nothing to do with our library code

File: smart-app/src/components/function/pick.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    const result = pick(obj, ['user.name','user.address.state']);
+    console.log(result);
```
[the remark this answers is not in the document]
Comment (bharat-dussa) [reply smart-app/src/components/function/pick.tsx:17]: @coderabbitai smart-app repo is for testing, this is nothing to do with our library code
